tests/test_pydantic_plugin.py

```diff
@@ -1266,3 +1267,39 @@ class MyModel(BaseModel):
 
     m = MyModel(x=1)
     assert cloudpickle.loads(cloudpickle.dumps(m)).model_dump() == m.model_dump() == {'x': 1}  # type: ignore
+
+
+def test_sqlmodel_pydantic_plugin(exporter: TestExporter) -> None:
+    logfire.instrument_pydantic()
```

Comment:
What in the `instrument_pydantic` causes this?

Reply:
I don't know.

Reply:
Looked into it more. In the `sqlmodel_validate` function, it calls `cls.__pydantic_validator__.validate_python` which triggers the plugin. That produces an object that's temporarily missing the `_sa_instance_state` attribute which causes `sa_inspect` to fail while the plugin is trying to log the object. Then `sqlmodel_validate` restores that attribute. So nothing stays broken and we shouldn't need to worry. In `logfire.info('x', x=Hero.model_validate({}))`, the span from the plugin doesn't have the right schema, but the `x` log does.